internal/clients/nightfall/nightfall.go

```diff
@@ -227,31 +251,53 @@ func (n *Client) ReviewDiff(ctx context.Context, logger logger.Logger, fileDiffs
 		}
 	}
 
-	comments := []*diffreviewer.Comment{}
-	// Integer round up division
-	numRequestsRequired := (len(contentToScanList) + maxItemsForAPIReq - 1) / maxItemsForAPIReq
-	logger.Debug(fmt.Sprintf("Sending %d requests to Nightfall API", numRequestsRequired))
-	for i := 0; i < numRequestsRequired; i++ {
-		// Use max number of items to determine content to send in request
-		cts := sliceListBySize(i, maxItemsForAPIReq, contentToScanList)
+	commentCh := make(chan []*diffreviewer.Comment)
+	newCtx, cancel := context.WithDeadline(ctx, time.Now().Add(time.Minute*20))
+	defer cancel()
 
-		// Pull out content strings for request
-		items := make([]string, len(cts))
-		for i, item := range cts {
-			items[i] = item.Content
-		}
+	go func() {
+		blockingCh := make(chan struct{}, n.MaxNumberRoutines)
+		var wg sync.WaitGroup
 
-		// send API request
-		resp, err := n.Scan(ctx, logger, items)
-		if err != nil {
-			logger.Debug(fmt.Sprintf("Error sending request number %d with %d items", i+1, len(items)))
-			return nil, err
+		// Integer round up division
+		numRequestsRequired := (len(contentToScanList) + maxItemsForAPIReq - 1) / maxItemsForAPIReq
+
+		logger.Debug(fmt.Sprintf("Sending %d requests to Nightfall API", numRequestsRequired))
+		for i := 0; i < numRequestsRequired; i++ {
+			// Use max number of items to determine content to send in request
+			contentSlice := sliceListBySize(i, maxItemsForAPIReq, contentToScanList)
+
+			wg.Add(1)
+			blockingCh <- struct{}{}
+			go func(loopCount int, cts []*contentToScan) {
+				defer wg.Done()
+				if newCtx.Err() != nil {
+					return
+				}
+
+				c, err := n.scanContent(newCtx, cts, loopCount+1, logger)
+				if err != nil {
+					cancel()
+				} else {
+					commentCh <- c
+				}
+				<-blockingCh
+			}(i, contentSlice)
 		}
+		wg.Wait()
+		close(commentCh)
```

Comment:
nit - i know there are currently no early exit points in this code block, but it seems like this call to `close` might be good to defer at the top of this go routine (could result in nasty bugs if later on someone adds to this func and ends up exiting early to handle an error or something)